{source_path} is empty

### Description of the problem

When setting the "Default Path" unter *Preferences > Output Files > Automatic File Naming* to `{source_path}/` and opening a source either by drag and drop or by selecting a source through the "Open Source" button, the "Save As" input field says `/name-of-file.mkv` instead of `C:/path/to/source/name-of-file.mkv`.

The file then gets created directly in `C:/`.

### HandBrake version (e.g., 1.0.0)

1.3.0 (2019110900)

### Operating system and version (e.g., Ubuntu 18.04 LTS, macOS 10.14 Mojave, Windows 10 1809)

Windows 10 Pro 1909 64-bit (10.0, Build 18363.535)

### Error message text or screenshot

N/A

### HandBrake Activity Log (see https://handbrake.fr/docs/en/latest/help/activity-log.html)

~~~
HandBrake 1.3.0 (2019110900)
OS: Microsoft Windows NT 10.0.18363.0
CPU: Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
Ram: 16310 MB, 
GPU Information:
  AMD Radeon R9 200 Series - 26.20.13001.50005
Screen: 1920x1080
Temp Dir: C:\Users\x3ro\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake
Data Dir: C:\Users\x3ro\AppData\Roaming\HandBrake

-------------------------------------------

[00:38:15] Compile-time hardening features are enabled
Cannot load nvEncodeAPI64.dll
[00:39:34] hb_init: starting libhb thread

 # Starting Scan ...

[00:39:34] CPU: Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
[00:39:34]  - Intel microarchitecture Haswell
[00:39:34]  - logical processor count: 4
[00:39:34] Intel Quick Sync Video support: no
[00:39:34] hb_scan: path=C:\Users\x3ro\Videos\Kwad FPV\2_5224635752020706470.mp4, title_index=0
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image C:\Users\x3ro\Videos\Kwad FPV\2_5224635752020706470.mp4
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(C:\Users\x3ro\Videos\Kwad FPV\2_5224635752020706470.mp4\) failed
[00:39:34] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[00:39:34] dvd: not a dvd - trying as a stream/file instead
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'C:\Users\x3ro\Videos\Kwad FPV\2_5224635752020706470.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    creation_time   : 2019-06-23T17:23:27.000000Z
  Duration: 00:00:05.42, start: 0.000000, bitrate: 2841 kb/s
    Stream #0:0(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 256 kb/s (default)
    Metadata:
      creation_time   : 2019-06-23T17:23:16.000000Z
      handler_name    : SoundHandle
    Stream #0:1(eng): Video: h264 (Baseline) (avc1 / 0x31637661), yuv420p, 1280x720, 2578 kb/s, 29.89 fps, 60 tbr, 90k tbn, 180k tbc (default)
    Metadata:
      creation_time   : 2019-06-23T17:23:16.000000Z
      handler_name    : VideoHandle
[00:39:34] scan: decoding previews for title 1
[00:39:34] scan: audio 0x0: aac, rate=48000Hz, bitrate=256163 English (AAC LC) (2.0 ch) (256 kbps)
[00:39:34] scan: 10 previews, 1280x720, 29.888 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1
[00:39:34] scan: supported video decoders: avcodec qsv
[00:39:34] libhb: scan thread found 1 valid title(s)

 # Processing Scan Information ...


 # Scan Finished ...


~~~

--- a/win/CS/HandBrakeWPF/Helpers/AutoNameHelper.cs
+++ b/win/CS/HandBrakeWPF/Helpers/AutoNameHelper.cs
@@ -196,7 +196,8 @@ namespace HandBrakeWPF.Helpers
 
         private static string GetAutonamePath(IUserSettingService userSettingService, EncodeTask task, string sourceName)
         {
-            string autoNamePath = userSettingService.GetUserSetting<string>(UserSettingConstants.AutoNamePath).Trim();
+            string autoNamePath = userSettingService.GetUserSetting<string>(UserSettingConstants.AutoNamePath).Trim()
+                .Replace("/", "\\");
 
             // If enabled, use the current Destination path.
             if (!userSettingService.GetUserSetting<bool>(UserSettingConstants.AlwaysUseDefaultPath) && !string.IsNullOrEmpty(task.Destination))
@@ -209,9 +210,9 @@ namespace HandBrakeWPF.Helpers
             }
 
             // Handle {source_path} 
-            if (userSettingService.GetUserSetting<string>(UserSettingConstants.AutoNamePath).Trim().StartsWith("{source_path}") && !string.IsNullOrEmpty(task.Source))
+            if (autoNamePath.StartsWith("{source_path}") && !string.IsNullOrEmpty(task.Source))
             {
-                string savedPath = userSettingService.GetUserSetting<string>(UserSettingConstants.AutoNamePath).Trim().Replace("{source_path}\\", string.Empty).Replace("{source_path}", string.Empty);
+                string savedPath = autoNamePath.Replace("{source_path}\\", string.Empty).Replace("{source_path}", string.Empty);
                 string directory = Directory.Exists(task.Source) ? task.Source : Path.GetDirectoryName(task.Source);
                 autoNamePath = Path.Combine(directory, savedPath);
             }

